### PR TITLE
a11y-tests: Mock Date class earlier

### LINKED ISSUE #10019: Calendar/DatePicker snapshot tests should not be specific to current month
There are some snapshot tests for the DatePicker/Calendar where the component is displaying the current or previous month, so they fail when the month changes. In [this example](https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=22470) it's the accessibility tests failing, but this may happen with the normal component examples snapshot test too.

image

We should find a way to prevent these tests from running on components showing the current or previous month.

cc: @lorejoh12 @statm

### PR DESCRIPTION
**Pull request by statm**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #10019
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

('Take two' of #9721)

There are a bunch of places, e.g. [Calendar](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx#L54), where we create `new Date()` in component's `defaultProps`, which happens earlier before mocking code in `beforeAll` kicks in during test. This will cause real `Date` class to leak into jest tests and eventually the monthly snapshot instability pointed out by #10019.

This PR moves mocking code for `Date` from `beforeAll` into Jest's setup file to ensure it executes before test code.

#### Focus areas to test

No weird date mismatches in future snapshot tests.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10029)

### CODE REVIEW

**Comment by ecraig12345** [comment]
Thanks for doing this! Looks like you need to update snapshots (for the last time on this issue hopefully!).

**Comment by ecraig12345** [comment]
Still failing, but this one doesn't make any sense... I wonder if the test is somehow time zone dependent as well?? (I noticed it failing on the build machines last night, but passing locally--maybe due to different time zones.)
image

**Comment by statm** [comment]
@ecraig12345 oh wow it's still failing... but good news is that after changing local time 2 months forward, there's no more failures. I'll investigate further to see how this field is generated.

File: apps/a11y-tests/config/tests.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ global.Date = class {
 
 // Ensure test output is consistent across machine locale and time zone config.
```
Comment (ecraig12345) [inline apps/a11y-tests/config/tests.js]: Should this comment be updated now? Maybe also add something about why you changed it.

Comment (ecraig12345) [reply apps/a11y-tests/config/tests.js]: Hmm also I think this variant will break if someone isn't in Pacific time... (and when DST starts/ends)

Comment (statm) [reply apps/a11y-tests/config/tests.js]: Good catch! Updated.

**Comment by statm** [comment]
Merging. We'll see if it churns again next month...

**Comment by ecraig12345** [comment]
> Merging. We'll see if it churns again next month...

I think now it might churn when someone updates snapshots from a different time zone...

**Comment by ecraig12345** [comment]
Hmm I changed my computer to central time, ran the update snapshots command, and it didn't do anything. So I guess this works?

**Comment by statm** [comment]
> Hmm I changed my computer to central time, ran the update snapshots command, and it didn't do anything. So I guess this works?

It's because all notions of UTC time are removed. Machines in different time zones will have different internal representation of `new Date(2017, 0, 6, 4, 41, 20)`, but that's okay - we only care about how they are displayed in DatePicker/Calendar, which are locale-aware.

**Comment by ecraig12345** [comment]
Oh got it, I didn't realize the date constructor worked on local time.

**Pull request merged**